modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -0,0 +1,582 @@
+
+public:
+
+    ONNXImporter(const char *onnxFile)
+    {
+        std::fstream input(onnxFile, std::ios::in | std::ios::binary);
+
+        if (!model_proto.ParseFromIstream(&input))
+            CV_Error(Error::StsUnsupportedFormat, "Failed to parse onnx model");
+    }
+
+    void populateNet(Net dstNet);
+};
+
+inline void replaceLayerParam(LayerParams& layerParams, const String& oldKey, const String& newKey)
+{
+    if (layerParams.has(oldKey)) {
+        layerParams.set(newKey, layerParams.get(oldKey));
+        layerParams.erase(oldKey);
+    }
+}
+
+void releaseONNXTensor(opencv_onnx::TensorProto& tensor_proto)
+{
+    if (!tensor_proto.raw_data().empty()) {
+        delete tensor_proto.release_raw_data();
+    }
+}
+
+Mat getMatFromTensor(opencv_onnx::TensorProto& tensor_proto)
+{
+    CV_Assert(!tensor_proto.raw_data().empty() || !tensor_proto.float_data().empty()
+                    || !tensor_proto.double_data().empty() || !tensor_proto.int64_data().empty());
+
+    opencv_onnx::TensorProto_DataType datatype = tensor_proto.data_type();
+    Mat blob;
+    std::vector<int> sizes;
+    for (int i = 0; i < tensor_proto.dims_size(); i++) {
+            sizes.push_back(tensor_proto.dims(i));
+    }
+    if (datatype == opencv_onnx::TensorProto_DataType_FLOAT) {
+
+        if (!tensor_proto.float_data().empty()) {
+            const ::google::protobuf::RepeatedField<float>& field = tensor_proto.float_data();
+            CV_Assert(!field.empty());
+            Mat(sizes, CV_32FC1, (void*)field.data()).copyTo(blob);
+        }
+        else {
+            char* val = const_cast<char*>(tensor_proto.raw_data().c_str());
+            Mat(sizes, CV_32FC1, val).copyTo(blob);
+        }
+    }
+    else if (datatype == opencv_onnx::TensorProto_DataType_DOUBLE)
+    {
+        const ::google::protobuf::RepeatedField<double>& field = tensor_proto.double_data();
+        CV_Assert(!field.empty());
+        Mat(sizes, CV_64FC1, (void*)field.data()).copyTo(blob);
+        blob.convertTo(blob, CV_32FC1);
+    }
+    else if (datatype == opencv_onnx::TensorProto_DataType_INT64)
+    {
+        blob.create(sizes, CV_32SC1);
+        int32_t* dst = reinterpret_cast<int32_t*>(blob.data);
+
+        if (!tensor_proto.int64_data().empty()) {
+            const ::google::protobuf::RepeatedField< ::google::protobuf::int64>& src = tensor_proto.int64_data();
+
+            for (int i = 0; i < blob.total(); i++) {
+                if (src[i] < std::numeric_limits<int32_t>::min() || src[i] > std::numeric_limits<int32_t>::max()) {
+                    CV_Error(Error::StsOutOfRange, "Input is out of OpenCV 32S range");
+                }
+                dst[i] = saturate_cast<int32_t>(src[i]);
+            }
+        }
+        else
+        {
+            char* val = const_cast<char*>(tensor_proto.raw_data().c_str());
+            int64_t* src = reinterpret_cast<int64_t*>(val);
+
+            for (int i = 0; i < blob.total(); i++) {
```

Comment:
It looks like we can just get a pointer to `int64` data inside an if-else condition but move similar conversion code outside it.